src/solver/conv_asm_implicit_gemm_gtc_bwd.cpp

```diff
@@ -0,0 +1,323 @@
+    };
+    // clang-format on
+    return tunables;
+}
+
+/*
+* Return with true if kernel is found, and its kernel_name, block_size, grid_size as well.
+* Return with false if no kernel can be executed with input ConvolutionContext
+*/
+static inline bool FindImplicitGemmGtcDynamicBwdKernel(const ConvolutionContext& ctx,
+                                                       std::string& kernel_name,
+                                                       int& block_size,
+                                                       int& grid_size)
+{
+    auto tunables  = GetImplicitGemmGtcDynamicBwdTunablesList();
+    auto pConfig   = tunables.begin();
+    int hi         = ctx.out_height;
+    int wi         = ctx.out_width;
+    int n          = ctx.batch_sz;
+    int k          = ctx.n_inputs;
+    int c          = ctx.n_outputs;
+    int ho         = ctx.in_height;
+    int wo         = ctx.in_width;
+    int stride_h   = ctx.in_height > 1 ? ctx.kernel_stride_h : 1;
+    int stride_w   = ctx.in_width > 1 ? ctx.kernel_stride_w : 1;
+    int dilation_h = ctx.kernel_size_h > 1 ? ctx.kernel_dilation_h : 1;
+    int dilation_w = ctx.kernel_size_w > 1 ? ctx.kernel_dilation_w : 1;
+    int pad_h      = ctx.pad_h;
+    int pad_w      = ctx.pad_w;
+    int y          = ctx.kernel_size_h;
+    int x          = ctx.kernel_size_w;
+
+    int gcd_stride_dilation_h = gcd(stride_h, dilation_h);
+    int gcd_stride_dilation_w = gcd(stride_w, dilation_w);
+    int y_tilda               = stride_h / gcd_stride_dilation_h;
+    int x_tilda               = stride_w / gcd_stride_dilation_w;
+
+    int h_tilda = ho + (dilation_h * (y - 1) + stride_h - 1) / stride_h;
+    int w_tilda = wo + (dilation_w * (x - 1) + stride_w - 1) / stride_w;
+
+    int y_dot = integer_divide_ceil(y, y_tilda);
+    int x_dot = integer_divide_ceil(x, x_tilda);
+
+    int h_tilda_left = std::max(0, pad_h - dilation_h * (y_tilda - 1)) / stride_h;
+    int w_tilda_left = std::max(0, pad_w - dilation_w * (x_tilda - 1)) / stride_w;
+
+    int h_tilda_right = std::min(h_tilda, (pad_h + hi - 1 + stride_h - 1) / stride_h + 1);
+    int w_tilda_right = std::min(w_tilda, (pad_w + wi - 1 + stride_w - 1) / stride_w + 1);
+
+    int h_tilda_slice = h_tilda_right - h_tilda_left;
+    int w_tilda_slice = w_tilda_right - w_tilda_left;
+    int num_of_gemm   = y_tilda * x_tilda;
+    // clang-format on
+    int gemm_m = c;
+    int gemm_n = n * h_tilda_slice * w_tilda_slice;
+
+    for(; pConfig != tunables.end(); pConfig++)
+    {
+        if(pConfig->nxe == 0)
+        {
+            if((x != 1) || (y != 1) || (stride_h != 1) || (stride_w != 1) || (dilation_h != 1) ||
+               (dilation_w != 1) || (pad_h != 0) || (pad_w != 0))
+            {
+                continue;
+            }
+        }
+        if((gemm_n % pConfig->gemm_n_per_block != 0) || (gemm_m % pConfig->gemm_m_per_block != 0) ||
+           (k % pConfig->gemm_k_per_block != 0))
```

Comment:
I think that it could be better to put gemm_k instead of k. And gemm_k = k*y*x. In some conv layers of mask rcnn y=x=2.

Reply:
yes, you are right, I'll fix it.